projects/storefrontlib/cms-components/misc/icon/icon.component.ts

```diff
@@ -41,87 +42,97 @@ export class IconComponent {
    * The cxIcon directive is bound to the icon type. You can feed the `ICON_TYPE` to
    * accomplish a configurable button in the UI.
    */
-  @Input() set cxIcon(type: ICON_TYPE) {
+  @Input() set cxIcon(type: ICON_TYPE_STRING) {
     this.setIcon(type);
   }
 
   /**
    * The type input parameter is bound to the icon type. You can feed the `ICON_TYPE` to
    * accomplish a configurable button in the UI.
    */
-  @Input() set type(type: ICON_TYPE) {
+  @Input() set type(type: ICON_TYPE_STRING) {
+    // XXX this clashes with the type attribute of some html form elements, e.g. input or button
     this.setIcon(type);
   }
 
-  /**
-   * the icon provides an html fragment that is used to add SVG or text based icons.
-   */
-  icon: SafeHtml | undefined;
-
-  /**
-   * The `flip-at-rtl` class is added to the DOM for the style layer to flip the icon in RTL direction.
-   */
-  @HostBinding('class.flip-at-rtl') flipAtRtl: boolean;
-
-  /**
-   * The `flip-at-ltr` class is added to the DOM for the style layer to flip the icon in LTR direction.
-   */
-  @HostBinding('class.flip-at-ltr') flipAtLtr: boolean;
-
   /**
    * Maintains the applied style classes so we can remove them when the
    * icon type changes at run time.
    */
-  protected styleClasses: string[];
+  @HostBinding('class') styleClasses: string[] = [];
+
+  private iconValue?: string;
+  private iconResourceType: IconResourceType;
 
   constructor(
     protected iconLoader: IconLoaderService,
-    protected elementRef: ElementRef<HTMLElement>,
-    protected renderer: Renderer2
+    protected sanitizer: DomSanitizer
   ) {}
 
-  protected setIcon(type: ICON_TYPE): void {
+  protected setIcon(type: ICON_TYPE_STRING): void {
     if (!type || <string>type === '') {
       return;
     }
-    this.icon = this.iconLoader.getHtml(type);
-    this.addStyleClasses(type);
+    this.iconResourceType = this.iconLoader.getResourceType(type);
+    if (this.isSvgIcon()) {
+      // looks like angular does not sanitize URLs when binding to the xlink:href attribute of svg elements
+      // so we need to sanitze the URL manually
+      this.iconValue =
+        this.sanitizer.sanitize(
+          SecurityContext.URL,
+          this.iconLoader.getSvgPath(type) || null
+        ) || undefined;
+    } else if (this.isTextIcon()) {
+      this.iconValue = this.iconLoader.getSymbol(type);
```

Comment:
Fyi, `TEXT` icons are not treated as pure text. If the value happens to contain html markup, it will not be executed by the browser. As a result:
* Storefront visitors will see any html markup from `TEXT` icons verbatim.
* `TEXT` icons will never cause XSS.